openapi2beans/pkg/generator/yaml2java.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package generator
+
+import (
+	"log"
+	"strings"
+
+	openapi2beans_errors "github.com/dev-galasa/buildutils/openapi2beans/pkg/errors"
+	"github.com/galasa-dev/cli/pkg/files"
+)
+
+func GenerateFiles(fs files.FileSystem, projectFilePath string, apiFilePath string, packageName string) error {
+	var fatalErr error
+	var apiyaml string
+	var errList map[string]error
+
+	storeFilePath := generateStoreFilePath(projectFilePath, packageName)
+	apiyaml, fatalErr = fs.ReadTextFile(apiFilePath)
+	if fatalErr == nil {
+		var schemaTypes map[string]*SchemaType
+		schemaTypes, errList, fatalErr = getSchemaTypesFromYaml([]byte(apiyaml))
+		if fatalErr == nil {
+			if len(errList) > 0 {
+				fatalErr = handleErrList(errList)
+			}
+			if fatalErr == nil {
+				fatalErr = generateDirectories(fs, storeFilePath)
+				if fatalErr == nil {
+					javaPackage := translateSchemaTypesToJavaPackage(schemaTypes, packageName)
+					convertJavaPackageToJavaFiles(javaPackage, fs, storeFilePath)
+				}
+			}
+		}		
+	}
+
+	return fatalErr
+}
+
+// Cleans and/or creates the store file
+func generateDirectories(fs files.FileSystem, storeFilePath string) error {
+	log.Println("Cleaning generated beans directory: " + storeFilePath)
+	exists, err := fs.DirExists(storeFilePath)
+	if err == nil {
+		if exists {
+			fs.DeleteDir(storeFilePath)
```

Comment:
Looks like it will still delete the entire source tree, rather than just the package you are generating into ?
If thats not the case, perhaps explain what the contents of storeFilePath is at this point ?